### PR TITLE
set title based on prefix and suffix in xresources

### PR DESCRIPTION
**Pull request by GRFreire**
The title is broken in a literal prefix and in a path suffix.
Both can be configured via X-resources.

https://github.com/muennich/sxiv/pull/453#issue-906753721
For those who use window managers (e.g. dwm, icewm) that show window titles, it may be useful to show the current file name in the title. This PR breaks the title in a string literal prefix and a suffix that shows a file path component, both X-resources configurable.

The a separator (" - ") in only set if needed.

```
sxiv - fileInView.png
^^^^^^^
prefix ^^^^^^^^^^^^^^
           suffix
```

The suffix can be the filename, its full path, its directory name, its directory full path, or, to keep backwards compatibility, nothing. Please see the manual for more detailed information.

### CODE REVIEW

**Comment by N-R-K** [comment]
Is this 1:1 https://github.com/muennich/sxiv/pull/453#issue-906753721 or did any change need to be made? If there was no change (no merge conflicts) then we should wget the patch from [here](https://github.com/muennich/sxiv/pull/453/commits/a099491b6182574b335acb196a37b613c7a6d080.patch) and use `git am` to merge it. It will preserve `author` and commit msg.

**Comment by XPhyro** [comment]
> Is this 1:1 [muennich/sxiv#453](https://github.com/muennich/sxiv/pull/453) or did any change need to be made? If there was no change (no merge conflicts) then we should wget the patch from [here](https://github.com/muennich/sxiv/pull/453/commits/a099491b6182574b335acb196a37b613c7a6d080.patch) and use `git am` to merge it. It will preserve `author` and commit msg.

It looks like there are some little changes, but I would still rather the patch be preserved and the diffs committed independently.

**Comment by eylles** [comment]
@qsmodo

**Comment by GRFreire** [comment]
I am not that expert on git, how could I get the author commit back and commit my changes separately. (now that i've already commited all together)

**Comment by N-R-K** [comment]
> It looks like there are some little changes, but I would still rather the patch be preserved and the diffs committed independently.

In that case I think the original author should be preserved and we should use `Co-authored-by:` for [co-authors](https://github.blog/2018-01-29-commit-together-with-co-authors/).

> I am not that expert on git, how could I get the author commit back and commit my changes separately. (now that i've already commited all together)

`git commit --amend --author="name <email@example.com>"`

Will change the author of the last commit. Then run `git commit --amend -v` to edit the commit message and put yourself as the co-author.

**Comment by N-R-K** [comment]
Let's set away from the author issue and focus on the patch itself for now.

- - -

Should this be added in? I can see this being useful in certain cases. But this does make a good amount of code complexity.

Also what are some usecases for wanting to have directory name in the title? I can't quite think of any.
I think we should limit the options to just, `empty`, `full-filename`, `base-filename`.

**Comment by eylles** [comment]
Pergapa viewing files from multiple directories, but even then `full-filename` is more descriptive.

**Comment by XPhyro** [comment]
> Also what are some usecases for wanting to have directory name in the title? I can't quite think of any.

The directory name could be useful in thumbnail mode.

**Comment by GRFreire** [comment]
Really think I messed things up here. Maybe close this PR and open another one with a clean commit from @qsmodo  then other for my changes.

**Comment by N-R-K** [comment]
We should rebase instead of merging from now on.

**Comment by N-R-K** [comment]
@GRFreire don't worry, I fixed it.

**Comment by GRFreire** [comment]
Seems like @N-R-K fixed my mess, thank you. now we can proceed with what matters.

**Comment by N-R-K** [comment]
> The directory name could be useful in thumbnail mode.

Aright, that's a valid use-case. However, we can avoid extra allocation if we don't allow dir names. And overall I think the patch would be much simpler.

**Comment by N-R-K** [comment]
@XPhyro Let's rebase instead of merging from now on. It's cleaner that way.
`git rebase branch-to-rebase-ontop-of`

**Comment by N-R-K** [comment]
@XPhyro there's a couple hardcoded `sxiv` in here. We'll need to change them on the rename PR later.

**Comment by N-R-K** [comment]
Anyways, lets hear some more opinion before proceeding with this patch.
I'm against allowing dir names as it would simplify the patch by quite a lot.

**Comment by XPhyro** [comment]
> @XPhyro there's a couple hardcoded `sxiv` in here. We'll need to change them on the rename PR later.

Will take care after this PR is resolved.

**Comment by GRFreire** [comment]
Looks like everyone agrees in having these options:
- Basename of file
- Full path to file
- Empty

About the directory (both basename and path) is not decided.

About that, I think that using it in thumbnail mode is useful, but the suffix mode is used both in thumbnail as in image mode, which I think directory in the title doesnt make much sense. I simply putting, I wont use this option so I dont mind removing it.

**Comment by XPhyro** [comment]
I don't have any strong opinion on this, so I'll just list some factors:

Pros to directories:
- Might have some use cases, especially in thumbnail mode.
- Makes the formatting options "complete".

Pros to no directories:
- Simplifies code.

**Comment by eylles** [comment]
i think deciding on this would benefit from knowing what rationale did @qsmodo had for including the directory options other than "formatting completeness"

**Comment by TAAPArthur** [comment]
@GRFreire 
One thing I would add is the ability to set title-prefix from the cli and probably also to config.def.h itself. Mainly for completeness but it would also help with launches. Like when some random application opens nsxiv knowing some context that can't be derived from the file path

**Comment by GRFreire** [comment]
> One thing I would add is the ability to set title-prefix from the cli and probably also to config.def.h itself. Mainly for completeness but it would also help with launches. Like when some random application opens nsxiv knowing some context that can't be derived from the file path

What the priority should be? Something like `cli > xresources > config.h > default`?

**Comment by GRFreire** [comment]
What flag (letter) do you think is best, since `-t` is already used. I'm adding the option for both title_prefix and title_suffixmode, so how would I proceed?

**note:** config.h options already implemented, just not sure how to go with the cli option yet.

**Comment by XPhyro** [comment]
`-N` is instance name, so I think `-T` makes sense.

**Comment by GRFreire** [comment]
`-T` for the prefix or suffixmode? What about the other? Or set both of them with the same tag (dont know how this would work out)

**Comment by XPhyro** [comment]
I was thinking the title would be passed as a single argument with formatting parameters.

**Comment by N-R-K** [comment]
`-T "suffix;prefix"`
This should be easy enough to parse.

**Comment by qsmodo** [comment]
> Is this 1:1 muennich/sxiv#453 or did any change need to be made? 

The PR is 1:1 with the latest commit in muennich/sxiv.

> Should this be added in? I can see this being useful in certain cases. But this does add some amount of code complexity.

Yes, several files were changed, and a POSIX header file was included, `libgen.h`.

> Also what are some usecases for wanting to have directory name in the title?

I often have two files with same name sitting in different sibling directories (i.e. `A/1.png` and `B/1.png`). Then it becomes essential to know at which one I'm looking.

"But we have the status-bar for that". True, the status-bar can already display that information if one tweaks `image-info`. Still, consider this:

- The window title may be a large space in some setups. E.g. it is a large part of e.g. DWM's status-bar, *which otherwise would go wasted* ([a screenshot that shows such a waste](https://lee-phillips.org/estadobarpage/typical.png)). Also some may use taskbars or rolled-up windows, to which similar considerations apply.

- Some users might stuff quite some information in the status-bar, so using the space in the title may be beneficial.

- Say you have Sxiv tiled in a small area of the screen, such that the whole path or directory does not fit the status-bar. Again, it may fill the title space.

If I were in your position, I might not be convinced those reasons are worth the code complexity, but I wanted it so I added to my fork. :)
You should feel free to drop it partially or entirely, adapt it, etc.

------

Off-topic: You guys move fast and circumstances changed a bit for me such that in the short term I'll have less time than I initially calculated to dedicate to this project. I'm sorry and hope I'll have more time in a not-so-distant future. Cheers.

**Comment by N-R-K** [comment]
> I often have two files with same name sitting in different sibling directories (i.e. `A/1.png` and `B/1.png`). Then it becomes essential to know at which one I'm looking.
>     * The window title may be a large space in some setups. E.g. it is a large part of e.g. DWM's status-bar, _which otherwise would go wasted_ ([a screenshot that shows such a waste](https://lee-phillips.org/estadobarpage/typical.png)). Also some may use taskbars or rolled-up windows, to which similar considerations apply.

Displaying full filepath should solve that.
 
I'm still not convinced that it's worth the code complexity to show directory names, the usecase of which is very limited and can be solved by displaying full filepath.

**Comment by XPhyro** [comment]
> > Is this 1:1 [muennich/sxiv#453](https://github.com/muennich/sxiv/pull/453) or did any change need to be made?
> 
> The PR is 1:1 with the latest commit in muennich/sxiv.

If I recall correctly, it was not when the PR opened. It got force pushed at some point.

> I often have two files with same name sitting in different sibling directories (i.e. `A/1.png` and `B/1.png`). Then it becomes essential to know at which one I'm looking.

That use case can be covered with the full path, no?
 
--------

> Off-topic: You guys move fast and circumstances changed a bit for me such that in the short term I'll have less time than I initially calculated to dedicate to this project. I'm sorry and hope I'll have more time in a not-so-distant future. Cheers.

We're all doing this for free here, don't force yourself :)

**Comment by eylles** [comment]
> 
> Off-topic: You guys move fast and circumstances changed a bit for me such that in the short term I'll have less time than I initially calculated to dedicate to this project. I'm sorry and hope I'll have more time in a not-so-distant future. Cheers.

Well we kinda have to move fast as the amount of work is rather sizeable, we still have to explore some options on what to do with stuff like the embedded icons, open issues from sxiv that we have to determine if they really fall within our scope or are for imlib2 to solve.

**Comment by qsmodo** [comment]
>        Is this 1:1 muennich/sxiv#453 or did any change need to be made?
>
>> The PR is 1:1 with the latest commit in muennich/sxiv.
>>
>>> If I recall correctly, it was not when the PR opened. It got force pushed at some point.

Some variable renaming; The existence separator is now handled by code logic, not by user direction via `prefix`; The XStoreName part was moved out of the function (since it only runs once it's fine).

> That use case can be covered with the full path, no?

If the only options are going to be either full path or nothing, then yes, the code can be quite simplified and one doesn't even need `libgen.h`. This looks like a very acceptable compromise.

On the other hand, if you want to display only the file name (not the path) in the window title — which some with limited space in window titles may want —, then you also need to offer `basename` as an option, and then you'll have the switch statement, so the other options are natural additional branches to it.

**Comment by XPhyro** [comment]
> Looks like everyone agrees in having these options:
> 
> * Basename of file
> * Full path to file
> * Empty

I thought we already had basename.

**Comment by GRFreire** [comment]
Now you can set this in your config.h
```c
/* default title prefix and suffix mode */
static const char *TITLE_PREFIX = "sxiv";
static const int TITLE_SUFFIXMODE = 0;
```

And in the cli you can have `sxiv -T "nsxiv;4"` (for example). If you only want to change the preffix, just put `-T nsxiv`, if what you want is the oposite (only suffixmode), simply put `-T ";4"`

**Comment by XPhyro** [comment]
> And in the cli you can have `sxiv -T "nsxiv;4"` (for example). If you only want to change the preffix, just put `-T nsxiv`, if what you want is the oposite (only suffixmode), simply put `-T ";4"`

I see you implemented it as `prefix;suffix`. I agree with @N-R-K that it should be `suffix;prefix`. This way, the prefix string can contain `;` without any ambiguity.

**Comment by qsmodo** [comment]
>     then you also need to offer basename as an option
>
>> I thought we already had basename.

Yes, what I meant is: If you are keeping _basename_, removing, or _path to directory_ and _directory basename_ simplifies the code because it removes two branches of the switch, but even more simplifying and perhaps justified would be to remove _file basename_: You could even do away with it`libgen.h`.

On second thought, I agree that keeping _path to directory_ and _directory basename_ is not justified. The only benefit is that one can track down the directory with a quick glance on the title and the file name with a glance on the status-bar, while the former would be more difficult if the full file path were displayed. Given this scenario is quite unlikely and specific, I think it is pretty justified remove the directory options.

File: window.c
Comment (XPhyro) [inline window.c:138]: Shouldn't we register .title.suffixmode here?

**Review by XPhyro — CHANGES_REQUESTED** [review]

File: window.c
Comment (GRFreire) [reply window.c:138]: Probably. Now that I am editing the code, I kept thinking that.

**Comment by GRFreire** [comment]
Looks like no one is against removing suffix modes 1 and 3 (directories) and some are in favor, so I'll take that as a consensus.

**Comment by XPhyro** [comment]
Currently `sxiv -T asdf path` assumes `suffixmode = 0`. Should `sxiv` die if `suffixmode` is invalid?

**Comment by GRFreire** [comment]
> Currently `sxiv -T asdf path` assumes `suffixmode = 0`. Should `sxiv` die and/or warn on stderr if `suffixmode` is invalid?

I think we could either warn that the -T arg is invalid, or assume that this is the prefix, not the suffix. I am leaning towards the first option, but idk.

**Review by XPhyro — DISMISSED** [review]

**Review by XPhyro — CHANGES_REQUESTED** [review]
The current parsing does not allow for setting the prefix to an empty string. Working on a fix right now.

**Comment by TAAPArthur** [comment]
@XPhyro 
> The current parsing does not allow for setting the prefix to an empty string. Working on a fix right now.

Wouldn't it be easier to just have to separate flags for both parts of the title?

**Comment by XPhyro** [comment]
> Wouldn't it be easier to just have to separate flags for both parts of the title?

It would have been, but this is cleaner in the CLI.

**Review by XPhyro — APPROVED** [review]

**Comment by N-R-K** [comment]
I wasn't a fan of this patch since I felt it was needlessly complex to begin with, but now the complexity has increased even more. Let's drop directory names and Xresources. For prefix allow only

* 0 - for an empty string
* 1 - for base filename
* 2 - for full filename

We can have a default in `config.h` and provide provide cli flags that overrides it.

**Comment by N-R-K** [comment]
The problem with `-T` is that smaller `t` is taken. What do you think about `-X` and `-x` for prefix and suffix? They can stand for "Xtitle". I think it's best if prefix and suffix have the same letter for the cli flag, makes it easy to remember. @XPhyro

**Comment by eylles** [comment]
i think `-X` and `-x` can work really well, they can even be interpreted as prefi`X` and suffi`x`

File: sxiv.1
Comment (N-R-K) [inline sxiv.1:412]: It's more natural to have 0 as empty, 1 as basename and 2 as full path.

File: sxiv.h
Comment (N-R-K) [inline sxiv.h:124]: What's the purpose of `SUFFIXMODE_COUNT`?
Also there's an empty line here. And the rest of the codebase **doesn't** use comma on the last enum.

File: window.c
Comment (N-R-K) [inline window.c:158]: Let's remove Xresources completely and just have config.h and cli flags. Users can use the `nsxiv-env` script to set default cli flags if they installed a binary version.

File: sxiv.h
Comment (N-R-K) [inline sxiv.h:430]: Do we need this? We already have `title_prefix` and `title_suffixmode` inside `opt`.

File: sxiv.h
Comment (N-R-K) [inline sxiv.h:292]: Shouldn't `title_suffixmode` be of `suffixmode_t` type?

File: window.c
Comment (N-R-K) [inline window.c:541]: This should be removed. If the user wants a seperator they can just append it to the end of prefix, i.e `"sxiv -"`.

**Review by N-R-K — CHANGES_REQUESTED** [review]

File: sxiv.1
Comment (XPhyro) [reply sxiv.1:412]: I think this was chosen such that 0 will be the default. I am okay with both arrangements.

Comment (N-R-K) [reply sxiv.1:412]: I think we should have default be empty, even if we want basename to be default, 0 should still correspond to empty. It's just much more natural.

- - -

Also let's have more readable enum names, `EMPTY`, `BASENAME`, `FULLPATH` would be nice.

**Comment by XPhyro** [comment]
I like the single argument better as these two options are very closely related. I think it's worth the hassle since the added complexity is only in argument parsing and not using.

Agree that Xresources and separator can be removed.

Also, if we're removing Xresources, we should `strtol` in `options.c` and die if suffixmode is invalid. Currently the behaviour is to only warn.

**Comment by N-R-K** [comment]
What argument format are we going for? `-T "prefix;suffix_mode"` ?

**Comment by XPhyro** [comment]
> What argument format are we going for? `-T "prefix;suffix_mode"` ?

`-T suffix_mode;prefix`

**Comment by N-R-K** [comment]
> > What argument format are we going for? `-T "prefix;suffix_mode"` ?
> 
> `-T suffix_mode;prefix`

Isn't that a bit backwards, I think prefix should come first. It's prefix after all.

**Comment by XPhyro** [comment]
> Isn't that a bit backwards, I think prefix should come first. It's prefix after all.

We split at the first `;` so this was easier to implement. But it looks like there is `strrchr` that is the reverse of `strchr`, so this should be just as easy to implement.

I'll change it.

**Comment by XPhyro** [comment]
@N-R-K Should I check for the suffixmode *not* being 0, 1 or 2? The other options do not check, but this way we get a segfault (which could be fixed by assuming a default).

File: options.c
Comment (N-R-K) [inline options.c:78]:
```suggestion
	while ((opt = getopt(argc, argv, "A:abce:fG:g:hin:N:opqrS:s:T:tvZz:")) != -1) {
```

File: options.c
Comment (N-R-K) [inline options.c:157]: Is optarg guaranteed to be null terminated?

Also if someone puts `-T ";a"` it should error out for invalid suffix, not set prefix as "a".  And `-T ";2"` should set prefix to be empty and suffix to be fullpath.

File: options.c
Comment (N-R-K) [inline options.c:161]: We may wanna check if there was anything after `;` before incrementing the pointer.

**Review by N-R-K — CHANGES_REQUESTED** [review]

**Comment by N-R-K** [comment]
I think we should just go for 2 different flags, makes things much simpler.

File: options.c
Comment (XPhyro) [reply options.c:157]: It is guaranteed to be null-terminated.

It used to error out. I didn't think of this in the format reversal.

File: options.c
Comment (XPhyro) [reply options.c:161]: It's guaranteed to be null-terminated, so if there is nothing there, it'll construct an empty string.

**Comment by XPhyro** [comment]
I agree that it will be much simpler, but I think this is cleaner in the CLI. The complexity of the parsing shouldn't be a problem after the initial merge.

**Comment by N-R-K** [comment]
I'm for having 2 seperate flags instead. Let's hear what others have to say.

File: window.c
Comment (N-R-K) [inline window.c:494]: A bit hacky, but I'm quite sure `/` is not allowed to be inside a filename so we should be fine.

One edge case I can think of is if path exceeds `PATH_MAX`. Not sure what happens then.

Comment (XPhyro) [reply window.c:494]: > A bit hacky, but I'm quite sure `/` is not allowed to be inside a filename so we should be fine.

Yes.

> One edge case I can think of is if path exceeds `PATH_MAX`. Not sure what happens then.

I don't think it is our problem if it exceeds `PATH_MAX`.

File: options.c
Comment (XPhyro) [inline options.c:157]:
```c
if ((s = strrchr(optarg, ';')) != NULL) {
	s[0] = '\0';
```

By the way, this is the only part related to parsing. The reversal simplified things quite a bit.

File: window.c
Comment (N-R-K) [reply window.c:494]: Anyways, don't think we should use PATH_MAX blindly. Should either malloc or use VLA.

**Comment by XPhyro** [comment]
> I'm for having 2 seperate flags instead. Let's hear what others have to say.
> I don't think it's a big deal having to do `-x "title" -X 2`, and we can always have shell wrapper for default args.

I like it as `-T`, but if the others agree, I'll conform.

> Also if we're trying to optimize of cli, then we shouldn't use `;` since that will have to be surrounded in quotes. Maybe use `:`.

Sure.

> Also what happens if someone wants to use the separator in his prefix?

It works fine.

**Comment by eylles** [comment]
If we go for consistency with the rest of the command line options of sxiv and the command line options if x11 programs then `-T` is the way to go, and yes the complexity is as always in te parsing process thankfully here it is just on the initial parsing.

**Comment by N-R-K** [comment]
@XPhyro update the docs too.

**Review by N-R-K — APPROVED** [review]

File: window.c
Comment (TAAPArthur) [inline window.c:495]: If path doesn't contain a "/", won't this segfault. `strrchr(path, '/')` would return null. I am assuming path is the path of the file relative to the current directly and not always an absolute path.

File: config.def.h
Comment (TAAPArthur) [inline config.def.h:16]: I don't think we need to guard around this. I don't think the existence of the feature actually hurts anyone who doesn't want to use it? If we keep them, I'd lean towards making it an opt-out feature. 

Actually upon further inspection, I think I'm missing the point. We define _TITLE_CONFIG in options.c before including so we benefit do we get to guarding this?

File: window.c
Comment (N-R-K) [reply window.c:495]: The path was acquired via `realpath` in `check_add_files` and checks for NULL was made. So it's practically impossible for `strrchr(path, '/')` to return NULL.

Comment (XPhyro) [reply window.c:495]: On my system, `path` is the full path. I do not know if this is system-dependent.

File: config.def.h
Comment (N-R-K) [reply config.def.h:16]: Just keeps things consistent with the rest of `config.def.h`. If you take another look, you'll notice that EVERYTHING is guarded.

Comment (XPhyro) [reply config.def.h:16]: This was standard practice from sxiv. The config file is chunked and the `#include`s only use the part they want to.

File: window.c
Comment (N-R-K) [reply window.c:495]: `realpath` will always return full path or NULL on failure. We already error out of failure in `check_add_files`, so this is perfectly fine, although a bit hacky.

Comment (eylles) [reply window.c:495]: welp, tested and tried real good to make it segfault but i couldn't so unless there's something else we can proceed with the approval

**Review by XPhyro — APPROVED** [review]

File: config.def.h
Comment (eylles) [reply config.def.h:16]: even if this isn't needed it is an artifact of the codebase for the sake of consistency, as such i'm marking it as resolved.

File: window.c
Comment (TAAPArthur) [reply window.c:495]: Ok cool. Well not cool. I think it is interesting they we use the full path and not the relative one. I feel like that would add a lot of extra noise as probably only a subset of the path is useful info. Well I don't have a immediate use case for the relative path, but the fact the absolute path should probably be highlighted in the docs as that isn't what I was expecting

File: config.def.h
Comment (TAAPArthur) [reply config.def.h:16]: Perhaps the consistency point is enough to drop the comment. But I don't understand why everything is guarded. It isn't easily possible to not compile these extra lines and doing so would break the program anyways. They just seem like glorified comments. I just want to make sure I'm not missing some deep reason for their existence.

**Comment by eylles** [comment]
just to make sure this all is expected behaviour before approval here's a table of commands and the title created by them.

|command|title|
|----|----|
|sxiv -T :0 image |`sxiv`|
|sxiv -T :1 image |`image`|
|sxiv -T :2 image |`/full/path/to/image`|
|sxiv -T sxiv: image|`sxiv`|
|sxiv -T sxiv:1 image|`sxivimage`|
|sxiv -T sxiv:2 image|`sxiv/full/path/to/image`|
|sxiv -T sxiv\  image|`sxiv image`|
|sxiv -T sxiv\ : image|`sxiv `|
|sxiv -T sxiv\ : image|`sxiv `|
|sxiv -T sxiv\ :1 image|`sxiv image`|
|sxiv -T sxiv\ :2 image|`sxiv /full/path/to/image`|
|sxiv -T "sxiv " image|`sxiv image`|
|sxiv -T "sxiv ": image|`sxiv `|
|sxiv -T "sxiv ": image|`sxiv `|
|sxiv -T "sxiv ":1 image|`sxiv image`|
|sxiv -T "sxiv ":2 image|`sxiv /full/path/to/image`|

and of course if you give `:"n"` to the -T option where `"n"` is any character other than a number from 0 to 2 it will error out with `sxiv: Invalid argument for option -T suffixmode: "n"`

**Comment by XPhyro** [comment]
I guess two commands might be of interest here:
```
sxiv -T :0    image # sxiv
sxiv -T sxiv: image # sxiv
```

1. For the first one, it defaults to `sxiv` because the whole title evaluates to an empty string.
2. For the second one, it allows `suffixmode` to be empty.

I think this behaviour is fine.

---

Also, `sxiv -T sxiv\ image` prints usage because no file was given, it does not have a title. `sxiv -T sxiv\ image image` would have the title `sxiv image`.

**Comment by TAAPArthur** [comment]
There is one case that seems to be missing from the above table
`sxiv -T sxiv image` (no ':') This causes the title to just be `sxiv` correct

**Comment by eylles** [comment]
> 
> Also, `sxiv -T sxiv\ image` prints usage because no file was given, it does not have a title. `sxiv -T sxiv\ image image` would have the title `sxiv image`.

my bad, i screwed up on that in the table as it should have had 2 spaces

**Review by TAAPArthur — APPROVED** [review]

**Review by eylles — APPROVED** [review]

**Comment by GRFreire** [comment]
Wow! You move so fast. Cant even sleep and there is 92 messages in this issue.

I cannot approve this pr since I am the author, but nicely done guys.

**Comment by N-R-K** [comment]
Wait why revert ?

**Comment by XPhyro** [comment]
Accidentally reverted. Undid the revert.

**Comment by N-R-K** [comment]
Should I clean up the commit msg?

**Comment by XPhyro** [comment]
If you're referring to `This reverts commit a9c78f8.`, the current commit is not that.

If you're referring to the actual squash message, I think it can stay as-is.

**Comment by N-R-K** [comment]
The commit msg also refers to Xresources, we dropped it. I think we should clean it up.

**Comment by GRFreire** [comment]
The commit message is `set title based on prefix and suffix in xresources` but we changed that and no longer get these from the xresources.

> Should I clean up the commit msg?

So yeah, probably.

**Comment by N-R-K** [comment]
Hmm, I can't force push.

**Comment by XPhyro** [comment]
Yeah, we should. I thought you meant the description.

**Comment by N-R-K** [comment]
Can you disable this protected thing for now. Enable it after I've pushed.

**Comment by XPhyro** [comment]
Should be enabled now.

**Comment by N-R-K** [comment]
Still can't. 
```
remote: error: GH006: Protected branch update failed for refs/heads/dev.
remote: error: At least 2 approving reviews are required by reviewers with write access.

```

**Comment by XPhyro** [comment]
Might be something else, I've enabled it in Settings/Branches. You too should have privileges to view/change those.

**Comment by XPhyro** [comment]
Wait, okay, it's asking for approvals. Removed the approval condition too, try now.

**Comment by N-R-K** [comment]
It went through.

**Comment by XPhyro** [comment]
It looks like we lost the squash description. Do we want it?

**Comment by eylles** [comment]
it ain't necesary, just that it lists the co authors is good enough, if someone wants a detailed rundown of what was done they can always look at this PR.

**Pull request merged**
